src/main/kotlin/com/pubnub/internal/v2/subscription/SubscriptionImpl.kt

```diff
@@ -0,0 +1,72 @@
+package com.pubnub.internal.v2.subscription
+
+import com.pubnub.api.PubNub
+import com.pubnub.api.models.consumer.pubsub.PNEvent
+import com.pubnub.api.v2.callbacks.EventEmitter
+import com.pubnub.api.v2.callbacks.EventListener
+import com.pubnub.api.v2.subscriptions.Subscription
+import com.pubnub.api.v2.subscriptions.SubscriptionCursor
+import com.pubnub.api.v2.subscriptions.SubscriptionOptions
+import com.pubnub.api.v2.subscriptions.SubscriptionSet
+import com.pubnub.internal.v2.callbacks.EventEmitterImpl
+import com.pubnub.internal.v2.entities.ChannelGroupName
+import com.pubnub.internal.v2.entities.ChannelName
+import org.jetbrains.annotations.TestOnly
+
+internal class SubscriptionImpl(
+    internal val pubnub: PubNub,
+    channels: Set<ChannelName> = emptySet(),
+    channelGroups: Set<ChannelGroupName> = emptySet(),
+    options: SubscriptionOptions = SubscriptionOptions.Default,
+) : Subscription(), EventEmitter {
+
+    internal val channels = channels.toSet()
+    internal val channelGroups = channelGroups.toSet()
+
+    private val filters = options.allOptions.filterIsInstance<Filter>()
+
+    @get:TestOnly
+    internal val eventEmitter = EventEmitterImpl(pubnub, this::accepts)
+
+    // helps deliver events only after subscribe was called, starting AT LEAST with the requested cursor.timetoken
```

Comment:
Maybe we could extend this comment or add new comment to "accepts" method explaining that when there is a new subscription with SubscriptionCursor then existing subscriptions will not respect new SubscriptionCursor.

I may be wrong here because in SubscribeCapable.kt I found:

     * Please note that passing a [cursor] to [subscribe] affects *all* subscriptions that are currently active in the
     * [com.pubnub.api.PubNub] client, as it will reset the global timetoken for the server connection.

Reply:
changed the wording in SubscribeCapable and here, PTAL